src/fileop.c

```diff
@@ -118,6 +177,7 @@ int librole_read_file_from_dir(const char * const directory,
 
     if (NULL == directory || NULL == filename || NULL == role_graph)
     {
+        retcode = LIBROLE_INCORRECT_VALUE;
```

Comment:
I suggest moving this assignment to line 170.

Reply:
It's not needed, default value not using in this function anymore. And LIBROLE_INCORRECT_VALUE not best choice for default error code.